spec/route_emitter_template_spec.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+# rubocop: disable Metrics/BlockLength
+require 'rspec'
+require 'json'
+require 'bosh/template/test'
+
+describe 'route_emitter' do
+  let(:release_path) { File.join(File.dirname(__FILE__), '..') }
+  let(:release) { Bosh::Template::Test::ReleaseDir.new(release_path) }
+  let(:job) { release.job('route_emitter') }
+
+  describe 'route_emitter.json.erb' do
+    let(:deployment_manifest_fragment) do
+      {
+        'bpm' => {
+          'enabled' => 'true'
+        },
+        'diego' => {
+          'route_emitter' => {
+            'bbs' => {
+              'ca_cert' => 'CA CERTS',
+              'client_cert' => 'CLIENT CERT',
+              'client_key' => 'CLIENT KEY'
+            },
+            'nats' => {
+              'ca_cert' => 'CA CERTS',
+              'client_cert' => 'CLIENT CERT',
+              'client_key' => 'CLIENT KEY',
+              'machines' => {
+                'nats_addresses' => '1.2.3.4'
+              }
+            },
+          }
+        },
+        'enable_consul_service_registration' => 'false',
+        'loggregator' => 'LOGGREGATOR PROPS',
+        'logging' => {
+          'format' => {
+            'timestamp' => 'rfc3339'
+          }
+        }
+      }
+    end
+
+    let(:template) { job.template('config/route_emitter.json') }
+    let(:rendered_template) { template.render(deployment_manifest_fragment) }
+
+    context 'Check jitter_factor default value' do
+      it 'Defaults to 0.2 if no value is provided' do
+        json_data = JSON.parse(rendered_template)
+        jitter_value = json_data['jitter_factor']
+        expect(jitter_value).to eq(0.2)
+      end
+    end
+
+    context 'Check if jitter_factor value is less than 1.0' do
```

Comment:
As mentioned earlier, a test is missing for when the value is not a float.
e.g.
```
      it 'fails if jitter_factor value is not a float' do
        deployment_manifest_fragment['diego']['route_emitter']['jitter_factor'] = 'banana'
        expect do
          rendered_template
        end.to raise_error(/diego.route_emitter.jitter_factor should not be less than 0.0 and should not be greater than 1.0/)
      end

```